app/models/course/announcement/settings.rb

```diff
@@ -0,0 +1,40 @@
+class Course::Announcement::Settings
```

Comment:
Opening another namespace seems extraneous to me. Your model and controller should correspond.